src/components/MultiCombobox/MultiCombobox.tsx

```diff
@@ -282,21 +298,37 @@ function MultiCombobox<Item>({
             }
 
             // Get clipboard data and split them based on newline and/or commas
-            const clipboardData = e.clipboardData.getData('Text');
-            const items = (clipboardData
-              .replace(/\r?\n/g, ',')
-              .split(',')
-              .map(processUserValue)
-              .filter(validateUserValue) as unknown) as Item[];
-
-            if (items.length > 1) {
-              // Prevent the text from actually being pasted to the underlying input
-              e.preventDefault();
-              e.stopPropagation();
-
-              // extend existing values with new ones
-              onChange([...value, ...items]);
+            const clipboardData = e.clipboardData.getData('Text').trim();
+
+            const setItems = (items: string[]) => {
+              if (items.length > 1) {
+                // Prevent the text from actually being pasted to the underlying input
+                e.preventDefault();
+                e.stopPropagation();
+
+                // extend existing values with new ones
+                onChange([
+                  ...value,
+                  ...(([...items]
+                    .map(processUserValue)
+                    .filter(validateUserValue) as unknown) as Item[]),
+                ]);
+              }
+            };
+
+            // convert JSON-style data to something prettier
+            if (clipboardData[0] === '{' || clipboardData[0] === '[') {
+              try {
+                const parsed = JSON.parse(clipboardData);
+
+                setItems(flattenObject(parsed));
+                return;
+              } catch (e) {
+                // no error -- user input just isn't a JSON object
+              }
             }
+
+            setItems(clipboardData.replace(/\r?\n/g, ',').split(','));
```

Comment:
I feel that we should remove this business logic from herree and just allow the user to specify it. I'd even go as far as remove the default `.replace(/\r?\n/g, ',').split(','))` and just have panther-enterprise specify this on a wrapper.

Thoughts?